sklearn/metrics/tests/test_common.py

```diff
@@ -1110,3 +1110,101 @@ def test_no_averaging_labels():
             score_labels = metric(y_true, y_pred, labels=labels, average=None)
             score = metric(y_true, y_pred, average=None)
             assert_array_equal(score_labels, score[inverse_labels])
+
+
+def test_metric_permutation_invariance_multiclass():
+    rng = np.random.RandomState(42)
+    n_classes = 5
+    n_samples = 100
+    p_C = rng.rand(n_classes)
+    p_C /= p_C.sum()
+    cp_C = np.cumsum(p_C)
+    y_true = np.searchsorted(cp_C, rng.rand(n_samples))
+    y_pred = []
+    for y in y_true:
+        if rng.randint(0, 10) > 5:
```

Comment:
why not just `rng.random() > .6`?

Indeed, why not just inserting noise at random: `y_pred = y_true.copy(); y_pred[rng.randint(len(y_true), size=n_samples // 3)] = np.random.randint(n_classes)`? Or `y_pred[rng.randint(len(y_true), size=n_samples // 3)] = y_true[rng.randint(len(y_true), size=n_samples // 3)]`.

As long as the data is non-trivial, I am only seeking that the code is quickly readable.